### ISSUE TITLE
Build godot2 math functions missing

### ISSUE THREAD

**Issue by albfan** [godotengine/godot#37672]
**Godot version:** 2.1.6

**OS/device including version:** Arch linux (just updated)

**Issue description:** Error building godot. Seems it miss some math functions in link step

**Steps to reproduce:**

```
  scons platform=x11 \
        tools=yes \
        target=release_debug \
        use_llvm=yes \
        builtin_openssl=yes \
        colored=yes \
        pulseaudio=no bits=64-j 4
```

```
scons: Reading SConscript files ...
Enabling ALSA
scons: done reading SConscript files.
scons: Building targets ...
[ 75%] Linking Static Library ==> editor/libeditor.x11.opt.tools.64.llvm.a
[ 83%] Ranlib Library         ==> editor/libeditor.x11.opt.tools.64.llvm.a
[100%] progress_finish(["progress_finish"], [])
[100%] Linking Program        ==> bin/godot.x11.opt.tools.64.llvm
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(mpc_demux.x11.opt.tools.64.llvm.o): in function `mpc_set_replay_level':
mpc_demux.c:(.text+0x3c5b): undefined reference to `__exp2_finite'
/bin/ld: mpc_demux.c:(.text+0x3cb6): undefined reference to `__exp2_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(streaminfo.x11.opt.tools.64.llvm.o): in function `streaminfo_read_header_sv7':
streaminfo.c:(.text+0x3d9): undefined reference to `__log10_finite'
/bin/ld: streaminfo.c:(.text+0x411): undefined reference to `__log10_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(nb_celp.x11.opt.tools.64.llvm.o): in function `nb_decoder_ctl':
nb_celp.c:(.text+0x782b): undefined reference to `__log_finite'
/bin/ld: nb_celp.c:(.text+0x7845): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(sb_celp.x11.opt.tools.64.llvm.o): in function `sb_encode':
sb_celp.c:(.text+0x1041): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(vbr.x11.opt.tools.64.llvm.o): in function `vbr_analysis':
vbr.c:(.text+0x300): undefined reference to `__log_finite'
/bin/ld: vbr.c:(.text+0x433): undefined reference to `__pow_finite'
/bin/ld: vbr.c:(.text+0x6a6): undefined reference to `__log_finite'
/bin/ld: vbr.c:(.text+0x6d0): undefined reference to `__log_finite'
/bin/ld: vbr.c:(.text+0x857): undefined reference to `__log_finite'
/bin/ld: vbr.c:(.text+0x8b9): undefined reference to `__log_finite'
/bin/ld: vbr.c:(.text+0x903): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(vbr.x11.opt.tools.64.llvm.o):vbr.c:(.text+0x950): more undefined references to `__log_finite' follow
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(webp_enc.x11.opt.tools.64.llvm.o): in function `WebPEncode':
webp_enc.c:(.text+0x844): undefined reference to `__log10_finite'
/bin/ld: webp_enc.c:(.text+0x8be): undefined reference to `__log10_finite'
/bin/ld: webp_enc.c:(.text+0x92a): undefined reference to `__log10_finite'
/bin/ld: webp_enc.c:(.text+0x998): undefined reference to `__log10_finite'
/bin/ld: webp_enc.c:(.text+0xa1e): undefined reference to `__log10_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(mdct.x11.opt.tools.64.llvm.o): in function `mdct_init':
mdct.c:(.text+0x58): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(psy.x11.opt.tools.64.llvm.o): in function `_vp_psy_init':
psy.c:(.text+0xc0): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x129): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x198): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x2ff): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x8d0): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x8e8): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x99f): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0xa99): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(psy.x11.opt.tools.64.llvm.o): in function `setup_tone_curves':
psy.c:(.text+0x15e6): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x1617): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x1656): undefined reference to `__log_finite'
/bin/ld: psy.c:(.text+0x1922): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x1944): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x1cd9): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x1cfa): undefined reference to `__exp_finite'
/bin/ld: psy.c:(.text+0x2071): undefined reference to `__exp_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(sharedbook.x11.opt.tools.64.llvm.o): in function `_float32_pack':
sharedbook.c:(.text+0x4d): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(sharedbook.x11.opt.tools.64.llvm.o): in function `_book_maptype1_quantvals':
sharedbook.c:(.text+0x427): undefined reference to `__pow_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(sharedbook.x11.opt.tools.64.llvm.o): in function `_book_unquantize':
sharedbook.c:(.text+0x722): undefined reference to `__pow_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(sharedbook.x11.opt.tools.64.llvm.o): in function `vorbis_book_init_encode':
sharedbook.c:(.text+0xa5d): undefined reference to `__pow_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(frame_enc.x11.opt.tools.64.llvm.o): in function `VP8EncLoop':
frame_enc.c:(.text+0x798): undefined reference to `__log10_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(frame_enc.x11.opt.tools.64.llvm.o): in function `VP8EncTokenLoop':
frame_enc.c:(.text+0x19c4): undefined reference to `__log10_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(picture_csp_enc.x11.opt.tools.64.llvm.o): in function `ImportYUVAFromRGBA':
picture_csp_enc.c:(.text+0xea9): undefined reference to `__pow_finite'
/bin/ld: picture_csp_enc.c:(.text+0xf6c): undefined reference to `__pow_finite'
/bin/ld: picture_csp_enc.c:(.text+0xfa8): undefined reference to `__pow_finite'
/bin/ld: picture_csp_enc.c:(.text+0x1009): undefined reference to `__pow_finite'
/bin/ld: picture_csp_enc.c:(.text+0x1050): undefined reference to `__pow_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(picture_csp_enc.x11.opt.tools.64.llvm.o):picture_csp_enc.c:(.text+0x10ad): more undefined references to `__pow_finite' follow
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(lsp.x11.opt.tools.64.llvm.o): in function `vorbis_lsp_to_curve':
lsp.c:(.text+0x52e): undefined reference to `__exp_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(lossless_enc.x11.opt.tools.64.llvm.o): in function `FastLog2Slow_C':
lossless_enc.c:(.text+0x184c): undefined reference to `__log_finite'
/bin/ld: modules/libmodules.x11.opt.tools.64.llvm.a(lossless_enc.x11.opt.tools.64.llvm.o): in function `FastSLog2Slow_C':
lossless_enc.c:(.text+0x1904): undefined reference to `__log_finite'
/bin/ld: drivers/libdrivers.x11.opt.tools.64.llvm.a(png.x11.opt.tools.64.llvm.o): in function `png_gamma_8bit_correct':
png.c:(.text+0x4267): undefined reference to `__pow_finite'
/bin/ld: drivers/libdrivers.x11.opt.tools.64.llvm.a(png.x11.opt.tools.64.llvm.o): in function `png_gamma_16bit_correct':
png.c:(.text+0x42b7): undefined reference to `__pow_finite'
/bin/ld: drivers/libdrivers.x11.opt.tools.64.llvm.a(png.x11.opt.tools.64.llvm.o): in function `png_gamma_correct':
png.c:(.text+0x4310): undefined reference to `__pow_finite'
/bin/ld: png.c:(.text+0x4353): undefined reference to `__pow_finite'
/bin/ld: drivers/libdrivers.x11.opt.tools.64.llvm.a(png.x11.opt.tools.64.llvm.o): in function `png_build_gamma_table':
png.c:(.text+0x47e1): undefined reference to `__pow_finite'
/bin/ld: drivers/libdrivers.x11.opt.tools.64.llvm.a(png.x11.opt.tools.64.llvm.o):png.c:(.text+0x4a01): more undefined references to `__pow_finite' follow
/bin/ld: editor/libeditor.x11.opt.tools.64.llvm.a(baked_light_baker.x11.opt.tools.64.llvm.o): in function `BakedLightBaker::_plot_light(BakedLightBaker::ThreadStack&, Vector3 const&, AABB const&, Color const&, Color const&, bool, Plane const&)':
baked_light_baker.cpp:(.text+0x6438): undefined reference to `__powf_finite'
/bin/ld: editor/libeditor.x11.opt.tools.64.llvm.a(baked_light_baker.x11.opt.tools.64.llvm.o): in function `BakedLightBaker::update_octree_sampler(DVector<int>&)':
baked_light_baker.cpp:(.text+0xb36f): undefined reference to `__pow_finite'
/bin/ld: baked_light_baker.cpp:(.text+0xb38b): undefined reference to `__pow_finite'
/bin/ld: baked_light_baker.cpp:(.text+0xb3a7): undefined reference to `__pow_finite'
/bin/ld: editor/libeditor.x11.opt.tools.64.llvm.a(baked_light_baker.x11.opt.tools.64.llvm.o): in function `BakedLightBaker::update_octree_images(DVector<unsigned char>&, DVector<unsigned char>&)':
baked_light_baker.cpp:(.text+0xba2b): undefined reference to `__powf_finite'
/bin/ld: baked_light_baker.cpp:(.text+0xba48): undefined reference to `__powf_finite'
/bin/ld: baked_light_baker.cpp:(.text+0xba65): undefined reference to `__powf_finite'
/bin/ld: editor/libeditor.x11.opt.tools.64.llvm.a(baked_light_baker.x11.opt.tools.64.llvm.o): in function `BakedLightBaker::_plot_pixel_to_lightmap(int, int, int, int, unsigned char*, Vector3 const&, Vector3 const&, double*, float, float)':
baked_light_baker.cpp:(.text._ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff[_ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff]+0x3d1): undefined reference to `__powf_finite'
/bin/ld: baked_light_baker.cpp:(.text._ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff[_ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff]+0x3f1): undefined reference to `__powf_finite'
/bin/ld: editor/libeditor.x11.opt.tools.64.llvm.a(baked_light_baker.x11.opt.tools.64.llvm.o):baked_light_baker.cpp:(.text._ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff[_ZN15BakedLightBaker23_plot_pixel_to_lightmapEiiiiPhRK7Vector3S3_Pdff]+0x40e): more undefined references to `__powf_finite' follow
/bin/ld: scene/libscene.x11.opt.tools.64.llvm.a(tween_interpolaters.x11.opt.tools.64.llvm.o): in function `expo::in(float, float, float, float)':
tween_interpolaters.cpp:(.text+0x5bf): undefined reference to `__exp2_finite'
/bin/ld: scene/libscene.x11.opt.tools.64.llvm.a(tween_interpolaters.x11.opt.tools.64.llvm.o): in function `expo::out(float, float, float, float)':
tween_interpolaters.cpp:(.text+0x62a): undefined reference to `__exp2_finite'
/bin/ld: scene/libscene.x11.opt.tools.64.llvm.a(tween_interpolaters.x11.opt.tools.64.llvm.o): in function `expo::in_out(float, float, float, float)':
tween_interpolaters.cpp:(.text+0x6c9): undefined reference to `__exp2_finite'
/bin/ld: tween_interpolaters.cpp:(.text+0x726): undefined reference to `__exp2_finite'
/bin/ld: scene/libscene.x11.opt.tools.64.llvm.a(tween_interpolaters.x11.opt.tools.64.llvm.o): in function `expo::out_in(float, float, float, float)':
tween_interpolaters.cpp:(.text+0x7e0): undefined reference to `__exp2_finite'
/bin/ld: scene/libscene.x11.opt.tools.64.llvm.a(tween_interpolaters.x11.opt.tools.64.llvm.o):tween_interpolaters.cpp:(.text+0x82a): more undefined references to `__exp2_finite' follow
/bin/ld: core/libcore.x11.opt.tools.64.llvm.a(math_funcs.x11.opt.tools.64.llvm.o): in function `Math::ease(double, double)':
math_funcs.cpp:(.text+0x476): undefined reference to `__pow_finite'
/bin/ld: math_funcs.cpp:(.text+0x4a9): undefined reference to `__pow_finite'
/bin/ld: math_funcs.cpp:(.text+0x4bd): undefined reference to `__pow_finite'
/bin/ld: math_funcs.cpp:(.text+0x4da): undefined reference to `__pow_finite'
/bin/ld: core/libcore.x11.opt.tools.64.llvm.a(math_funcs.x11.opt.tools.64.llvm.o): in function `Math::pow(double, double)':
math_funcs.cpp:(.text+0x501): undefined reference to `__pow_finite'
/bin/ld: core/libcore.x11.opt.tools.64.llvm.a(math_funcs.x11.opt.tools.64.llvm.o): in function `Math::log(double)':
math_funcs.cpp:(.text+0x7e1): undefined reference to `__log_finite'
/bin/ld: core/libcore.x11.opt.tools.64.llvm.a(math_funcs.x11.opt.tools.64.llvm.o): in function `Math::exp(double)':
math_funcs.cpp:(.text+0x7f1): undefined reference to `__exp_finite'
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.x11.opt.tools.64.llvm] Error 1
scons: building terminated because of errors.
```

**Minimal reproduction project:**

Build from 2.1.6 tar.gz

**Comment by akien-mga** [comment]
>         pulseaudio=no bits=64-j 4

Is this a typo? Or are you passing `bits="64-j"`? That might mess up some code that queries `env["bits"]` to conditionally enable things.

**Comment by albfan** [comment]
@akien-mga is a typo.

I think this is related with glibc 2.31.0

https://bugs.llvm.org/show_bug.cgi?id=45034

**Comment by akien-mga** [comment]
> I think this is related with glibc 2.31.0
>
> https://bugs.llvm.org/show_bug.cgi?id=45034

If you're on latest Arch Linux, you shouldn't be having this issue though, it has been fixed before the release of LLVM 10.0.0.
What's the output of `clang --version`?

**Comment by akien-mga** [comment]
> If you're on latest Arch Linux, you shouldn't be having this issue though, it has been fixed before the release of LLVM 10.0.0.
> What's the output of `clang --version`?

Ah nevermind, Arch Linux still ships with LLVM 9.0.1. So much for being cutting edge ;P

**Comment by albfan** [comment]
```
$ clang --version
clang version 9.0.1 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

as soon as I'm on 10 I will give feedback
